packages/engine/crawlers/solana/collectibles.ts

```diff
@@ -70,11 +74,13 @@ export const addCollectibleToState = async (
 		collectionMetadata?.address.toString() ||
 		nft.collection?.address.toString() ||
 		nft.address.toString();
-	const collectionId = `${address}/${collectionAddress}`;
+	const collectionId = `${address}/collection/${collectionAddress}`;
+	const collectibleId = `${address}/collectible/${nft.mint.address.toString()}`;
 
 	const collection = collectionState.map.get(collectionId);
+	const collectible = collectibleState.map.get(collectibleId);
```

Comment:
for now, I think we just need to use the storage, decouple from state. That means we will not use state at engine level anymore. It will make logical flow cleaner, just update storage. Any storage change will affect to app state by storage listener.
this comment also means we need to refactor this engine by just using storage

Reply:
@fluctlight-kayaba pls have a look